File: dest/18_abstract.swift
class Base {
  func f() {}
  func g() {
    print("g() called")
  }
  init() {
  }
}

class Derived: Base {
  override func f() {
    fatalError("Method is abstract")
  }
  override init() {
  }
}

class Derived2: Base {
  override func f() {
    print("f() called")
  }
  override init() {
  }
}

func main(args: [String]) {
  Derived2().f()
  Derived2().g()
}

main([])

File: test-src/18_abstract.kt
open class Base {
  open fun f() {}
  fun g() {
    println("g() called")
  }
}

abstract class Derived : Base() {
  override abstract fun f()
}

class Derived2 : Base() {
  override fun f() {
    println("f() called")
  }
}

fun main(args: Array<String>) {
  Derived2().f()
  Derived2().g()
}
